js/request.js

```diff
@@ -64,19 +63,24 @@ var startJob = function (job, requestObject, handler, opts) {
   var event = requestObject.event;
 
   var context = makeContextObject(opts.timeout, requestObject.context || {});
-  context.done = function (err, result) {
-    job.doCompletion(err, result);
-    if (err) {
-      console.warn('Error:', err);
-    }
+
+  const done = (err, res) => {
+      if (err) {
+          console.warn('Error:', err);
+          job.doError(err);
+      } else {
+          job.doCompletion(err, res);
+      }
```

Comment:
This looks to be changing the semantics - from *always* running `doCompletion` to only running on success; which means that `doCompletion` is only ever called with a null `err`.